gtest build fails due to outside array bound access in test_bitops

### Describe the bug
```
src/ucs/arch/bitops.h:186:14: error: array subscript 20 is outside array bounds of 'uint8_t [20]' {aka 'unsigned char [20]'} [-Werror=array-bounds]
  186 |     return ((*((uint8_t*)ptr1 + length) & ~UCS_MASK(8 - remainder)) ==
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~
test/gtest/ucs/test_bitops.cc:87:13: note: while referencing 'buffer1'
   87 |     uint8_t buffer1[20] = {0};
```

In test_ops.cc, there are cases where the buffer size is 20, and then there are assertions like `ASSERT_TRUE(ucs_bitwise_is_equal(buffer1, buffer2, 160));` which will result to accessing memory beyond the buffer allocation in ucs_bitwise_is_equal(), because 160 will result in an offset equal to 20.

The issue shows up when building with gcc 10.

### Steps to Reproduce
- `./contrib/configure-devel --prefix=SOME_PATH --enable-mt && make -j install`
- UCX master branch at 034b2389
- `gcc (GCC) 10.3.0`

--- a/src/ucs/arch/bitops.h
+++ b/src/ucs/arch/bitops.h
@@ -182,6 +182,10 @@ ucs_bitwise_is_equal(const void *ptr1, const void *ptr2, uint64_t bit_length)
         return 0;
     }
 
+    if (remainder == 0) {
+        return 1;
+    }
+
     /* Compare up to 7 last bits */
     return ((*((uint8_t*)ptr1 + length) & ~UCS_MASK(8 - remainder)) ==
             (*((uint8_t*)ptr2 + length) & ~UCS_MASK(8 - remainder)));

